client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx

```diff
@@ -0,0 +1,76 @@
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import { useTranslate } from 'i18n-calypso';
+import React, { useEffect, useMemo, useState } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+import { useSelector } from 'calypso/state';
+import { Plugin } from 'calypso/state/plugins/installed/types';
+import { getPluginActionStatuses } from '../plugin-management-v2/utils/get-plugin-action-statuses';
+import { useActions } from './use-actions';
+import { useFields } from './use-fields';
+
+import './style.scss';
+
+interface Props {
+	currentPlugins: Array< Plugin >;
+	initialSearch?: string;
+	isLoading: boolean;
+	onSearch?: ( search: string ) => void;
+	bulkActionDialog: ( action: string, plugins: Array< Plugin > ) => void;
+}
+
+export default function PluginsListDataViews( {
+	currentPlugins,
+	initialSearch,
+	isLoading,
+	onSearch,
+	bulkActionDialog,
+}: Props ) {
+	const allStatuses = useSelector( getPluginActionStatuses );
+	const translate = useTranslate();
+
+	const fields = useFields( bulkActionDialog );
+	const actions = useActions( bulkActionDialog );
+
+	const [ dataViewsState, setDataViewsState ] = useState< DataViewsState >( () => ( {
+		...initialDataViewsState,
+		perPage: 15,
+		search: initialSearch,
+		fields: [ 'plugins', 'sites', 'update' ],
+	} ) );
+
+	// When search changes, notify the parent component
+	useEffect( () => {
+		onSearch && onSearch( dataViewsState.search || '' );
+	}, [ dataViewsState.search, onSearch ] );
+
+	const { data, paginationInfo } = useMemo( () => {
+		const result = filterSortAndPaginate( currentPlugins, dataViewsState, fields );
+
+		// Add all statuses to the plugin object so we can display them in the DataViews
+		result.data.forEach( ( plugin ) => {
+			plugin.allStatuses = allStatuses.filter( ( status ) => status.pluginId === plugin.id );
```

Comment:
I've noticed we create `allStatuses` in this component only to pass it down to `PluginActionStatus` via the [`plugins` field](https://github.com/Automattic/wp-calypso/blob/04e61486e434796d0da8518c874ec3949d4f3c9d/client/my-sites/plugins/plugins-list/use-fields.tsx#L54). Can we retrieve the statuses in `PluginActionStatus` directly?

Reply:
Agreed. Also, intersecting with my comment about `getPluginActionStatuses`, it seems like there's a missed opportunity for some selector to efficiently package all necessary state into something judiciously keyed for easy retrieval (key by status?)

Reply:
Thanks for instigating this improvement. I figured out how to remove `getPluginActionStatuses` and how to still have the correct statuses at the plugin. [Here's the fix](https://github.com/Automattic/wp-calypso/pull/95561/commits/12c6d2f0d5727162e9a523a2dba45878f89df554).